### PR TITLE
Remove dependencies and refactor test code and data

### PR DESCRIPTION
**Pull request by jcrivenaes**
* Remove the need for `xtgeo-testdata` repo
* Use of fixtures relating to pytest `tmp_path` (i.e. no longer `TMP` folder)
* Remove some `xtgeo` dependecies

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import os
+import shutil
+from pathlib import Path
+
+import pytest
+
+
+@pytest.fixture(name="sourcepath", scope="session", autouse=True)
+def fixture_sourcepath():
```
Comment (berland) [inline tests/conftest.py:9]: Is this warranted as a fixture? Compared to defining testroot just as a constant in each test module?

(fixtures make it harder to read, unless the fixture here really provides some extra feature I don't see)

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    print("Current path is ", testroot)
+    return testroot
+
+
+@pytest.fixture(name="datatree", scope="session", autouse=True)
+def fixture_datatree(tmp_path_factory):
+    """Create a tmp folder structure for testing."""
+    tmppath = tmp_path_factory.mktemp("grd3dmaps")
+
+    # tmppath.mkdir(parents=True, exist_ok=True)
```
Comment (berland) [inline tests/conftest.py:21]: purge

File: tests/test_grid3d_average_map1.py [hunk 3]
```diff
@@ -1,74 +1,124 @@
...
-
 
-def test_average_map1a():
+def test_average_map1a(datatree):
     """Test HC thickness with YAML config example 1a ECL based"""
-    os.chdir(str(Path(__file__).absolute().parent.parent))
-    xx.main(["--config", "tests/yaml/avg1a.yml"])
+    result = datatree / "map1a_folder"
+    result.mkdir(parents=True)
+    xx.main(
```
Comment (berland) [inline tests/test_grid3d_average_map1.py:13]: Replace `xx` with the real name?

File: tests/test_grid3d_average_map1.py [hunk 4]
```diff
@@ -1,74 +1,124 @@
...
+    )
+
+    z1poro = xtgeo.surface_from_file(result / "z1--avg1b_average_por.gri")
     assert z1poro.values.mean() == pytest.approx(0.1598, abs=0.001)
     assert z1poro.values.std() == pytest.approx(0.04, abs=0.003)
 
-    imgs = glob.glob(os.path.join(TMPD, "*avg1b*.png"))
-    for img in imgs:
-        shutil.copy2(img, "docs/test_images/.")
+    # for auto documentation
```
Comment (berland) [inline tests/test_grid3d_average_map1.py:47]: Hmm, maybe flag this in function name and/docstring that this test function is not only a test function, but also critical for the documentation?

File: tests/test_grid3d_average_map1.py [hunk 5]
```diff
@@ -1,74 +1,124 @@
...
-def test_average_map1c():
+def test_average_map1c(datatree):
     """Test HC thickness with YAML config example 1c ROFF based, rotated map"""
-    os.chdir(str(Path(__file__).absolute().parent.parent))
-    xx.main(["--config", "tests/yaml/avg1c.yml"])
-
-    z1poro = xtgeo.surface_from_file(os.path.join(TMPD, "all--avg1c_average_por.gri"))
+    result = datatree / "map1c_folder"
+    result.mkdir(parents=True)
+    xx.main(
```
Comment (berland) [inline tests/test_grid3d_average_map1.py:56]: ditto for `xx` again (and for all `xx` and `xxx` below)

File: tests/test_grid3d_hc_thickness1.py [hunk 6]
```diff
@@ -1,143 +1,213 @@
...
+        [
+            "--config",
+            "tests/yaml/hc_thickness1b.yml",
+            "--mapfolder",
+            str(result),
+            "--plotfolder",
+            str(result),
+        ]
+    )
+    for img in result.glob("*hc1b*.png"):
```
Comment (berland) [inline tests/test_grid3d_hc_thickness1.py:51]: Flag documentation in function name/docstring

File: tests/yaml/hc_thickness3a.yml [hunk 7]
```diff
@@ -48,17 +44,18 @@ mapsettings:
...
-  diffvaluerange: [-3, 0]     # value range min/max for date diffs
+  valuerange:
+    [0, 10] # value range min/max (values above below
+    # will be truncated in plot!)
+  diffvaluerange: [-3, 0] # value range min/max for date diffs
 
-  xlabelrotation: 25      # Rotate the X axis labels (if they may overlap)
+  xlabelrotation: 25 # Rotate the X axis labels (if they may overlap)
 
-  colortable: rainbow     # colours jet/rainbow/seismic/gnuplot/gnuplot2/...
```
Comment (berland) [inline tests/yaml/hc_thickness3a.yml:60]: viridis possible?

File: tests/conftest.py (hunk 1 above)
Comment (jcrivenaes) [reply tests/conftest.py:9]: You're right, overkill here

File: tests/yaml/hc_thickness3a.yml (hunk 7 above)
Comment (jcrivenaes) [reply tests/yaml/hc_thickness3a.yml:60]: Will check

Comment (jcrivenaes) [reply tests/yaml/hc_thickness3a.yml:60]: `UserWarning: Trying to access as color map not installed in this version of matplotlib: <viridis>. Revert to <rainbow>
`
Seems that viridis is missing in many installations

**Review by berland — APPROVED** [review]
Good

**Pull request merged**